src/Corax/Querying/IndexSearcher.Search.cs

```diff
@@ -90,7 +93,33 @@ public IQueryMatch SearchQuery(FieldMetadata field, IEnumerable<string> values,
                     Constants.Search.Operator.And => And<IQueryMatch, MultiTermMatch>(searchQuery, query, token: cancellationToken),
                     _ => throw new ArgumentOutOfRangeException(nameof(@operator), @operator, null)
                 };
+                
+                continue;
+            }
+            
+            //Phrase query
+            var startPosition = terms.Count;
+            EncodeAndApplyAnalyzerForMultipleTerms(field, word, ref terms);
+            if (terms.Count == startPosition) continue; //sentence contained only stop-words
+
+            var hs = new HashSet<Slice>(SliceComparer.Instance);
+            for (var i = startPosition; i < terms.Count; ++i)
+            {
+                hs.Add(terms[i]);
             }
+
+            var allIn = AllInQuery(field, hs, cancellationToken: cancellationToken);
+
+            var phraseMatch = PhraseQuery(allIn, field, terms.ToSpan().Slice(startPosition, terms.Count - startPosition));
+
+            searchQuery = (searchQuery, @operator) switch
+            {
+                (null, _) => phraseMatch,
+                (_, Constants.Search.Operator.And) => And(searchQuery, phraseMatch, cancellationToken),
+                (_, Constants.Search.Operator.Or) => Or(searchQuery, phraseMatch, cancellationToken),
```

Comment:
What does those mean?

Reply:
@maciejaszyk ping

Reply:
This is handling operator part of `Search`
eg query like like `search(Field, ' word1 "word 2 word3" word4 ', OPERATOR (AND/OR))` has to be in AST like this:
`(Field == 'word1') OPERATOR (Field == 'word2 word3') OPERATOR (Field = 'word4')` 

We're building phrase as:
`PhraseMatch( AllIn(['word2', 'word3']). 'word2 word3')`

So, if `searchQuery` (our root) is not null we have append phrase query accordingly to the operator user has chosen `And(root, phraseQuery)`.